From a0a11078d842860c5aef1ee691935eb37c9758e4 Mon Sep 17 00:00:00 2001
From: Nishad Saraf <nishad.saraf@xilinx.com>
Date: Tue, 6 Oct 2020 00:35:38 -0700
Subject: [PATCH 1694/1851] misc: xilinx-ai-engine: Return NULL if no errors
 were found

commit a7b192fa61bad991e0a4bb1e1792c9f71a8c9c46 from
https://github.com/Xilinx/linux-xlnx.git

Return NULL if no errors were found in the local event resource
in aie_get_errors() API.

Signed-off-by: Nishad Saraf <nishad.saraf@xilinx.com>
Reviewed-by: Wendy Liang <wendy.liang@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/misc/xilinx-ai-engine/ai-engine-interrupt.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/misc/xilinx-ai-engine/ai-engine-interrupt.c b/drivers/misc/xilinx-ai-engine/ai-engine-interrupt.c
index 3e0444bc13d3..0df060f4df96 100644
--- a/drivers/misc/xilinx-ai-engine/ai-engine-interrupt.c
+++ b/drivers/misc/xilinx-ai-engine/ai-engine-interrupt.c
@@ -1068,13 +1068,18 @@ struct aie_errors *aie_get_errors(struct device *dev)
 		return ERR_PTR(ret);
 	}
 
+	num_errs = aie_get_error_count(apart);
+	if (!num_errs) {
+		mutex_unlock(&apart->mlock);
+		return NULL;
+	}
+
 	aie_errs = kzalloc(sizeof(*aie_errs), GFP_KERNEL);
 	if (!aie_errs) {
 		mutex_unlock(&apart->mlock);
 		return ERR_PTR(-ENOMEM);
 	}
 
-	num_errs = aie_get_error_count(apart);
 	error = kcalloc(num_errs, sizeof(*error), GFP_KERNEL);
 	if (!error) {
 		kfree(aie_errs);
-- 
2.31.1

